spec.html

```diff
@@ -45897,13 +45889,12 @@ <h1>
         <dl class="header">
         </dl>
         <emu-alg>
-          1. Let _taRecord_ be ? ValidateTypedArray(_typedArray_, ~unordered~).
           1. NOTE: Bounds checking is not a synchronizing operation when _typedArray_'s backing buffer is a growable SharedArrayBuffer.
+          1. Let _taRecord_ be ? ValidateTypedArray(_typedArray_, ~unordered~).
           1. If _waitable_ is *true*, then
             1. If _typedArray_.[[TypedArrayName]] is neither *"Int32Array"* nor *"BigInt64Array"*, throw a *TypeError* exception.
-          1. Else,
-            1. Let _type_ be TypedArrayElementType(_typedArray_).
-            1. If IsUnclampedIntegerElementType(_type_) is *false* and IsBigIntElementType(_type_) is *false*, throw a *TypeError* exception.
+          1. Let _type_ be TypedArrayElementType(_typedArray_).
```

Comment:
Why remove the `else` guard? There's no reason to do this check when `_waitable_` is `*true*`, since it is implied by the checks in that branch.